### PR TITLE
Integration refactor

### PR DESCRIPTION
**Pull request by mxaly**
* split and re organize integration code
* make additional integrations (like router) optional 
* prepare for more integrations

### CODE REVIEW

**Comment by mxaly** [comment]
@nekath @Szeliga @marcinkornek requesting for review :)

File: docs/api.md [hunk 1]
```diff
@@ -1,152 +1,155 @@
...
-  ```js
-  var element = document.getElementById('my-element');
-  renderComponent('MyComponentName', {foo: 'bar'}, element);
-  ```
+    ##### example:
 
-* ### unmountComponent
+    ```js
+    var element = document.getElementById('my-element');
+    renderComponent('MyComponentName', {ųfoo: 'bar'}, element);
```
Comment (nekath) [inline docs/api.md]: typo there is only one foo :)

File: docs/api.md [hunk 2]
```diff
@@ -1,152 +1,155 @@
...
-  ```
+    ```ruby
+    react_router(String router_name)
+    ```
+
+    ##### example:
+
+    ```ruby
+    <%= react_router('MyRouterName') %>
+    ```
```
Comment (nekath) [inline docs/api.md]: Imho we should use full path to method.
RWR.react.getComponent

**Comment by nekath** [comment]
I love namespaces, we should not be selfish and populate global namespace too much :)

But I don't want to type too much while using core methods.
This PR is important to discuss much more since it changes whole API.

File: docs/api.md (hunk 2 above)
Comment (mxaly) [inline docs/api.md]: Good point. I was thinking about it as weel.

**Pull request merged**